### ISSUE TITLE
Basic Auth config

### ISSUE THREAD

**Issue by rocketnova** [ralphbean/bugwarrior#295]
I'm trying to use bugwarrior with a redmine that sits behind a Basic Auth. I couldn't find any documentation for how to configure the .bugwarriorrc to handle that. Is this possible?

Thanks!

**Comment by ralphbean** [comment]
Quite possibly not currently possible.  ;)  But we could look into it!

I don't have access to a redmine instance behind Basic Auth to test with.  Is yours a semi-public instance where we could test?

**Comment by coddingtonbear** [comment]
Glancing at the source, it *looks* like we could probably swap out urllib2 for requests (which makes using basic auth a really trivial thing) with minimal effort?

**Comment by rocketnova** [comment]
Sadly, my redmine is not publicly available for testing against. Looking at the source, it looked possible to add an optional layer to handle basic auth in addition to the redmine api key. I actually hacked around in my source to do a proof of concept, which worked, but I'm not sure what the best and most secure way is to handle basic auth and also what the best way to add a config variable that gets picked up at the redmine service level.

**Comment by rocketnova** [comment]
Hi there,

I've made a pushed a proof of concept change to my fork at https://github.com/rocketnova/bugwarrior/commit/e55371e39ddc87e1ca8d9628fa6583495f1127b5. I'm not very experienced in python, so this is probably **not** the right solution, but this is currently working for what I need. Can something like this be incorporated?

Also, I noticed that 8516f968c5e0d51914c7710b722f1e00cda50b4b introduced a comma at the end of line 142 in  bugwarrior/services/__init__.py. This was totally throwing off my code, so I got rid of it. Is it intentional?

Thanks!
Rocket
